File: CS/ConsoleLogger.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.Xpo.Logger;

namespace S138124 {
    public class ConsoleLogger: ILogger {
        #region ILogger Members

        public void ClearLog() { }

        public int Count {
            get { return 0; }
        }

        public bool Enabled {
            get {
                return true;
            }
            set {
            }
        }

        public void Log(LogMessage message) {
            if(message.MessageType != LogMessageType.DbCommand) return;
            Console.WriteLine(message.MessageText);
        }

        public int LogLimit {
            get {
                return int.MaxValue;
            }
            set {
            }
        }

        public int LostMessageCount {
            get { return 0; }
        }

        public bool ServerActive {
            get { return true; }
        }

        #endregion
    }
}

File: VB/ConsoleLogger.vb
﻿Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.Xpo.Logger

Namespace S138124
	Public Class ConsoleLogger
		Implements ILogger
		#Region "ILogger Members"

		Public Sub ClearLog() Implements ILogger.ClearLog
		End Sub

		Public ReadOnly Property Count() As Integer Implements ILogger.Count
			Get
				Return 0
			End Get
		End Property

		Public Property Enabled() As Boolean Implements ILogger.Enabled
			Get
				Return True
			End Get
			Set(ByVal value As Boolean)
			End Set
		End Property

		Public Sub Log(ByVal message As LogMessage) Implements ILogger.Log
			If message.MessageType <> LogMessageType.DbCommand Then
				Return
			End If
			Console.WriteLine(message.MessageText)
		End Sub

		Public Property LogLimit() As Integer Implements ILogger.LogLimit
			Get
				Return Integer.MaxValue
			End Get
			Set(ByVal value As Integer)
			End Set
		End Property

		Public ReadOnly Property LostMessageCount() As Integer Implements ILogger.LostMessageCount
			Get
				Return 0
			End Get
		End Property

		Public ReadOnly Property ServerActive() As Boolean Implements ILogger.ServerActive
			Get
				Return True
			End Get
		End Property

		#End Region
	End Class
End Namespace
